### ISSUE TITLE
Leveraging glimpseRuntime

### ISSUE THREAD

**Issue by avanderhoorn** [Glimpse/Glimpse.Node.Prototype#5]
The idea behind `glimpseRuntime` is that it centralizes the point where we know about the beginning/ending of a request. From there, we can publish messages, execute other providers who want to know about it, etc. At the moment the middleware is publishing these messages directly which makes switching over to different web providers (i.e. express vs connect vs http, etc) more difficult. Any provider that comes online should simply have to call the begin/end functions in the runtime and everything should just work.

**Comment by mike-kaufman** [comment]
@avanderhoorn  - not sure how you're thinking this will work exactly.  Take a look at my latest check-in, and let me know if you have feedback.  In particular, how I hooked up the logic in express to detect when a route is being used, and raise the "action-route" message.  How I did this was to "over-ride" a specific method on the express Route object.  Not clear to me that there's a way to do this just from calling "beginRequest".
